### PR TITLE
fix issue#210: delete token.TotalSupply

### PR DESCRIPTION
**Pull request by ylsGit**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/token/genesis.go [hunk 1]
```diff
@@ -49,9 +49,9 @@ func defaultGenesisStateOKT() types.Token {
 		OriginalSymbol:      common.NativeToken,
 		WholeName:           "OKT",
 		OriginalTotalSupply: totalSupply,
-		TotalSupply:         totalSupply,
-		Owner:               addr,
-		Mintable:            true,
+		//TotalSupply:         totalSupply,
```
Comment (zhongqiuwood) [inline x/token/genesis.go:52]: remove this line

File: x/token/genesis_test.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ func TestInitGenesis(t *testing.T) {
 	newCtx := newMapp.BaseApp.NewContext(false, abci.Header{})
 
 	exportGenesis.Tokens[0].OriginalTotalSupply = sdk.NewDec(66666)
-	exportGenesis.Tokens[0].TotalSupply = sdk.NewDec(66666)
+	//exportGenesis.Tokens[0].TotalSupply = sdk.NewDec(66666)
```
Comment (zhongqiuwood) [inline x/token/genesis_test.go:85]: remove

File: x/token/genesis_test.go [hunk 3]
```diff
@@ -149,9 +149,9 @@ func TestIssueToken(t *testing.T) {
 		OriginalSymbol:      genesisState.Tokens[0].OriginalSymbol,
 		WholeName:           genesisState.Tokens[0].WholeName,
 		OriginalTotalSupply: genesisState.Tokens[0].OriginalTotalSupply,
-		TotalSupply:         genesisState.Tokens[0].TotalSupply,
-		Owner:               genesisState.Tokens[0].Owner,
-		Mintable:            genesisState.Tokens[0].Mintable,
+		//TotalSupply:         genesisState.Tokens[0].TotalSupply,
```
Comment (zhongqiuwood) [inline x/token/genesis_test.go:152]: remove

File: x/token/types/util.go [hunk 4]
```diff
@@ -13,11 +13,11 @@ import (
 )
 
 var (
-	notAllowedPrefix = "poolswap"
```
Comment (zhongqiuwood) [inline x/token/types/util.go:16]: do not introduce unnecessary changing

File: x/token/types/token.go [hunk 5]
```diff
@@ -7,15 +7,15 @@ import (
...
-	TotalSupply         sdk.Dec        `json:"total_supply" v2:"total_supply"`                   // e.g. 1000000000.00000000
-	Type                int            `json:"type"`                                             //e.g. 1 common token, 2 interest token
-	Owner               sdk.AccAddress `json:"owner" v2:"owner"`                                 // e.g. okchain1upyg3vl6vqaxqvzts69zpus2c027p7paw63s99
-	Mintable            bool           `json:"mintable" v2:"mintable"`                           // e.g. false
+	Description         string  `json:"description" v2:"description"`                     // e.g. "OK Group Global Utility Token"
+	Symbol              string  `json:"symbol" v2:"symbol"`                               // e.g. "okt"
+	OriginalSymbol      string  `json:"original_symbol" v2:"original_symbol"`             // e.g. "OKT"
+	WholeName           string  `json:"whole_name" v2:"whole_name"`                       // e.g. "OKT"
+	OriginalTotalSupply sdk.Dec `json:"original_total_supply" v2:"original_total_supply"` // e.g. 1000000000.00000000
+	//TotalSupply         sdk.Dec        `json:"total_supply" v2:"total_supply"`                   // e.g. 1000000000.00000000
```
Comment (zhongqiuwood) [inline x/token/types/token.go:15]: remove this line

File: x/token/types/token.go [hunk 6]
```diff
@@ -26,7 +25,27 @@ func (token Token) String() string {
 	return string(b)
 }
 
-type Tokens []Token
+type TokenResp struct {
```
Comment (ilovers) [inline x/token/types/token.go:28]:
```suggestion
type TokenResponse struct {
```

File: x/token/types/util.go [hunk 7]
```diff
@@ -169,3 +169,16 @@ func MergeCoinInfo(availableCoins, lockedCoins sdk.DecCoins) (coinsInfo CoinsInf
 	sort.Sort(coinsInfo)
 	return coinsInfo
 }
+
+func GenTokenResp(token Token) TokenResp {
```
Comment (ilovers) [inline x/token/types/util.go:173]:
```suggestion
func GenTokenResponse(token Token) TokenResp {
```

File: x/token/types/token.go [hunk 8]
```diff
@@ -26,7 +25,27 @@ func (token Token) String() string {
...
+
+func (token TokenResp) String() string {
+	b, err := json.Marshal(token)
+	if err != nil {
+		return "{}"
+	}
+	return string(b)
+}
+
+type Tokens []TokenResp
```
Comment (xiangjianmeng) [inline x/token/types/token.go:48]: Why Tokens is the slice of TokenResp

**Pull request merged**